### ISSUE TITLE
provide more documentation about the different classes and modules and their interaction

### ISSUE THREAD

**Issue by tillmo** [ontohub/ontohub#1291]
While reviewing some code, I realise that such a documentation is necessary on order not to lose the overall overview (and it is unrealistic to expect that everyone reads all the code).

**Comment by eugenk** [comment]
I assume you refer to https://github.com/ontohub/ontohub/pull/1263/files#r27343197 and https://github.com/ontohub/ontohub/pull/1263/files#r27343583, right?

**Comment by tillmo** [comment]
yes, for example. But these perhaps also could be documented locally. However, in general the structure of Ontohub has been quite complex and should be explained.
